activestorage/lib/active_storage.rb

```diff
@@ -67,6 +67,7 @@ module ActiveStorage
 
   mattr_accessor :replace_on_assign_to_many, default: false
   mattr_accessor :track_variants, default: false
+  mattr_accessor :old_signed_id_fallback, default: true
```

Comment:
Configuration booleans that opt into new behaviour is typically defaulted to false, and explicitly enabled with true.  This makes the logic switches much more readable and idiomatic. Maybe `mattr_accessor :direct_upload_signing`?

Reply:
Sounds fair, although I'm not sure about the name `direct_upload_purpose_signing` since we are always going to use the new purpose for signing, the flag is there only to stop allowing the old purpose (blob_id) as a fallback. What do you think of `ignore_old_purpose_for_direct_uploads` ? By default it would be `false` so we accept both: `blob_id` and `direct_upload` as the purpose.

Reply:
Hm, yeah my suggested name isn't quite right.  How about `ignore_blob_id_purpose_for_direct_uploads ` or maybe `find_direct_uploads_by_direct_upload_purpose`? Either of the these are fine with me.